### PR TITLE
xbps-src: Explain python module dependencies in Manual.md

### PR DESCRIPTION
**Pull request by ahesford**
Proper handling of python module dependencies in xbps-src needs some clarification. This is a first attempt.

cc: @ericonr

### CODE REVIEW

File: Manual.md [hunk 1]
```diff
@@ -1444,9 +1444,25 @@ Python packages should be built with the `python{,2,3}-module` build style, if p
...
 
-To allow cross compilation, the `python-devel` package (for python 2.7) must be added
-to `hostmakedepends` and `makedepends`. If any other python version is also supported,
-for example python3.4, those must also be added as host and target build dependencies.
+If a python package includes a compiled extension, the `python-devel` (for python 2.7) or
+`python3-devel` (for python 3.X) packages should be added to `makedepends`. Generally,
+dependencies on `python-*` or `python3-*` packages should be included in `hostmakedepends` because
+the package build process often requires the host interpreter be run with the necessary packages
+available. In most cases, especially those with pure-python dependencies, it is not necessary to
+duplicate these in `makedepends`. Exceptions are dependencies that provide both a python modul and
```
Comment (ericonr) [inline Manual.md:1452]: "python module"

File: Manual.md [hunk 2]
```diff
@@ -1444,9 +1444,25 @@ Python packages should be built with the `python{,2,3}-module` build style, if p
...
 building a python module for multiple versions from a single template is also possible.
 
-To allow cross compilation, the `python-devel` package (for python 2.7) must be added
-to `hostmakedepends` and `makedepends`. If any other python version is also supported,
-for example python3.4, those must also be added as host and target build dependencies.
+If a python package includes a compiled extension, the `python-devel` (for python 2.7) or
+`python3-devel` (for python 3.X) packages should be added to `makedepends`. Generally,
+dependencies on `python-*` or `python3-*` packages should be included in `hostmakedepends` because
+the package build process often requires the host interpreter be run with the necessary packages
+available. In most cases, especially those with pure-python dependencies, it is not necessary to
```
Comment (ericonr) [inline Manual.md:1451]: Can we make it clear that these should be the same dependencies that would also be in `depends`? I think that's what's needed, right?

**Comment by Chocimier** [comment]
Besides mention of pip, this is mainly repetition of "About the many types of depends variable". Mention of pip is too wordy, something like "If you notice build installing modules, add them to aproppriate `depends` variable. This is a case when ..." should suffice.

File: Manual.md (hunk 2 above)
Comment (ahesford) [reply Manual.md:1451]: The `setuptools` installer recognizes `install_requires`, which should generally be placed in `hostmakedepends`, and `requires`, which should be placed in `depends`. I'm going to rewrite some of this based on comments from @Chocimier to simplify and clarify this point.

File: Manual.md (hunk 1 above)
Comment (ahesford) [reply Manual.md:1452]: Obsolete

**Comment by ahesford** [comment]
Substantially shortened in the latest push.

File: Manual.md [hunk 3]
```diff
@@ -1444,9 +1444,18 @@ Python packages should be built with the `python{,2,3}-module` build style, if p
 This sets some environment variables required to allow cross compilation. Support to allow
 building a python module for multiple versions from a single template is also possible.
 
-To allow cross compilation, the `python-devel` package (for python 2.7) must be added
-to `hostmakedepends` and `makedepends`. If any other python version is also supported,
-for example python3.4, those must also be added as host and target build dependencies.
+Python packages that rely on `python-setuptools` or `python3-setuptools` should generally map
+`install_requires` dependencies in `setup.py` to `depends` in the template, and `setup_requires`
```
Comment (Chocimier) [inline Manual.md:1448]: While you are here, could you add: ".. `depends` in the template, or `python3` if there are none."?

**Comment by ahesford** [comment]
Removed the python2 mentions to keep the text a little cleaner and addressed request from @Chocimier.

**Comment by ahesford** [comment]
Closed in ff61ec45aeba3b19ba25a782037bc80e5a6e2182
